feat: React Query 전역 상태 관리 및 공통 fetcher 유틸 추가 (#78)

jngmnj opened the pull request:
## 🔗 관련 이슈

- Closes #78

## ✨ 작업 개요
* React Query 전역 상태 관리를 위한 환경 구성
* 공통 fetcher 유틸리티 함수 작성 및 인증/에러 처리 로직 반영
* 클라이언트/서버 모두에서 사용할 수 있는 안정적인 API 호출 구조 확보

## ✅ 작업 상세 내용
### 1. `Providers` 컴포넌트 추가 및 `layout.tsx` 적용

* `@tanstack/react-query` 설치 및 초기 세팅 완료
* `QueryClientProvider`, `ReactQueryDevtools` 포함된 `Providers` 컴포넌트 작성 (`use client`)
* `layout.tsx`에서 `Providers`로 전체 앱 감싸기 적용

### 2. 공통 fetcher 유틸 생성 및 구조화

* `fetcher<T>()` 함수 작성

  * `API_BASE_URL` 환경변수 기반으로 endpoint 구성
  * `accessToken` 존재 시 자동 Authorization 헤더 삽입
  * `POST`, `PUT` 요청 시에만 `Content-Type` 설정
  * 응답 실패 시 `ApiError` 클래스를 통해 상태코드 및 메시지 전달
  * 사용자 정의 headers 병합 처리

## 스크린샷
## 기타
* 후속 작업으로 fetcher를 사용하는 `queryFn` 모듈화 및 `useQuery` 적용 예정
* 추후 서버 전용 API 경로 분리 시, 서버 전용 환경변수(`INTERNAL_API_URL`) 도입 고려

jngmnj committed:
feat: 공통 fetcher 유틸 생성 및 인증/에러 처리 로직 추가 (#78)

- API_BASE_URL 환경변수 기반으로 endpoint 구성
- accessToken 존재 시 Authorization 헤더 자동 추가
- POST/PUT 등의 요청일 경우에만 Content-Type 자동 지정
- 응답 에러 발생 시 ApiError 클래스로 에러 정보 구조화
- 사용자가 전달한 headers는 병합되도록 처리

jngmnj committed:
feat: QueryClientProvider를 포함한 Providers 컴포넌트 추가 및 layout.tsx 적용 (#78)

- tanstack-query 추가
- React Query 전역 상태 관리를 위한 Providers 컴포넌트 생성
- QueryClientProvider 및 ReactQueryDevtools 포함
- layout.tsx에서 전체 애플리케이션을 Providers로 감싸도록 수정

Aprasaks commented on src/app/provider.tsx:
```diff
@@ -0,0 +1,16 @@
+import React, { useState } from 'react';
+
+export default function Providers({ children }: { children: React.ReactNode }) {
+  const [queryClient] = useState(() => new QueryClient());
```
기본설정값 안바꿔도 괜찮을까요? 
refetchonwindow랑  retry정도는 1번으로 바꿔도 괜찮을꺼같아요

jngmnj replied:
오 기본설정 추가하겠습니다!

jngmnj committed:
refactor: QueryClient 설정 외부 파일로 분리 및 provider 파일명 변경 (#78)

- queryClientOptions를 constants/api/queryClientOptions.ts로 분리
- QueryClientProvider 생성 시 옵션 적용 (staleTime, retry, refetch 등)
- provider.tsx 파일명을 providers.tsx로 변경하고 layout.tsx에서 경로 수정
